src/backend/storage/lmgr/deadlock.c

```diff
@@ -503,13 +524,29 @@ FindLockCycleRecurse(PGPROC *checkProc,
 	Assert(nVisitedProcs < MaxBackends);
 	visitedProcs[nVisitedProcs++] = checkProc;
 
+	/*
+	 * Remotexact
+	 * If this process is not the first we are visiting, and we are checking
+	 * for potential deadlocks involving a remotexact proc, then we return true
+	 * the first time we encounter another remotexact in the path.
+	 */
+	if (remote_start_proc && nVisitedProcs > 0) {
+		if (checkProc->isRemoteXact) {
+			ereport(LOG,
+					errmsg("Found a potential multi-region deadlock with"
+					"starting proc: %d and chain proc %d.", 
+					visitedProcs[0]->pid, checkProc->pid));
```

Comment:
The deadlock is already reported with `DeadlockReport()`. Is this logging necessary?

Reply:
I will add a TODO to remove it. I thought it we get too many aborts after adding this code; this line might be useful to determine where we are failing. I will remove it once we run some basic experiments.